### Import the nescessary libraly

In [1]:
import pandas as pd 
import numpy as np
from scipy import stats

Load the data

In [2]:
data = pd.read_csv('DataSet/K6dWbZwmSXCnVm2cJklwng_867ace16e19e49ec88e68cc191e66af1_c4_epa_air_quality.csv')
data.head(10)

Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   
5           5  2018-01-01        Hawaii      Honolulu  Not in a city   
6           6  2018-01-01        Hawaii      Honolulu  Not in a city   
7           7  2018-01-01  Pennsylvania          Erie           Erie   
8           8  2018-01-01        Hawaii      Honolulu       Honolulu   
9           9  2018-01-01      Colorado       Larimer   Fort Collins   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   
5                                            Kapolei  Carbon monoxide   
6                                            Kapolei  Carbon monoxide   
7                                                NaN  Carbon monoxide   
8                                           Honolulu  Carbon monoxide   
9                      Fort Collins - CSU - S. Mason  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3  
5  Parts per million         0.994737   14  
6  Parts per million         0.200000    2  
7  Parts per million         0.200000    2  
8  Parts per million         0.400000    5  
9  Parts per million         0.300000    6

Data Exploration

In [3]:
data.shape

(260, 10)

the dataset has 206 record and 10 columns

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 260 entries, 0 to 259
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        260 non-null    int64  
 1   date_local        260 non-null    object 
 2   state_name        260 non-null    object 
 3   county_name       260 non-null    object 
 4   city_name         260 non-null    object 
 5   local_site_name   257 non-null    object 
 6   parameter_name    260 non-null    object 
 7   units_of_measure  260 non-null    object 
 8   arithmetic_mean   260 non-null    float64
 9   aqi               260 non-null    int64  
dtypes: float64(1), int64(2), object(7)
memory usage: 20.4+ KB


`Dataset has 3 null value in the local_site_name column`

In [5]:
data.describe()

Unnamed: 0  arithmetic_mean         aqi
count  260.000000       260.000000  260.000000
mean   129.500000         0.403169    6.757692
std     75.199734         0.317902    7.061707
min      0.000000         0.000000    0.000000
25%     64.750000         0.200000    2.000000
50%    129.500000         0.276315    5.000000
75%    194.250000         0.516009    9.000000
max    259.000000         1.921053   50.000000

the `aqi` column has maximun value is 50 and the minium value is 0

In [7]:
data['county_name'].value_counts()

county_name
Los Angeles       14
Maricopa           8
Wayne              8
San Bernardino     6
Pima               5
                  ..
Burleigh           1
Imperial           1
New Haven          1
Mesa               1
Dodge              1
Name: count, Length: 149, dtype: int64

the dataset has 149 county_name distinct and Los Angeles is most frequency

In [10]:
data['state_name'].value_counts()

state_name
California              66
Arizona                 14
Ohio                    12
Florida                 12
Texas                   10
New York                10
Pennsylvania            10
Michigan                 9
Colorado                 9
Minnesota                7
New Jersey               6
Indiana                  5
North Carolina           4
Massachusetts            4
Maryland                 4
Oklahoma                 4
Virginia                 4
Nevada                   4
Connecticut              4
Kentucky                 3
Missouri                 3
Wyoming                  3
Iowa                     3
Hawaii                   3
Utah                     3
Vermont                  3
Illinois                 3
New Hampshire            2
District Of Columbia     2
New Mexico               2
Montana                  2
Oregon                   2
Alaska                   2
Georgia                  2
Washington               2
Idaho                    2
Nebraska         

In [16]:
data['state_name'].unique()

array(['Arizona', 'Ohio', 'Wyoming', 'Pennsylvania', 'Iowa', 'Hawaii',
       'Colorado', 'Minnesota', 'Indiana', 'Missouri', 'Connecticut',
       'California', 'Nevada', 'Virginia', 'Rhode Island', 'Texas',
       'Illinois', 'Kansas', 'Maryland', 'New Jersey', 'Massachusetts',
       'Maine', 'Idaho', 'Florida', 'Vermont', 'Tennessee', 'Georgia',
       'South Dakota', 'North Carolina', 'West Virginia', 'Michigan',
       'Nebraska', 'Alabama', 'Washington', 'Kentucky', 'Oklahoma',
       'New York', 'North Dakota', 'Montana', 'Utah', 'Delaware',
       'New Hampshire', 'Louisiana', 'Mississippi', 'New Mexico',
       'Oregon', 'District Of Columbia', 'Arkansas', 'Puerto Rico',
       'South Carolina', 'Alaska', 'Wisconsin'], dtype=object)

In [17]:
data['state_name'].nunique()

52

the dataset has 52 states distinct

### Statistical Tests

1. Formulate the null hypothesis and the alternative hypothesis.<br>
2. Set the significance level.<br>
3. Determine the appropriate test procedure.<br>
4. Compute the p-value.<br>
5. Draw your conclusion.

`ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.`

In [18]:
ca_los = data[data['county_name'] == 'Los Angeles']
ca_other = data[(data['state_name']=='California') & (data['county_name']!='Los Angeles')]

Formulate your hypothesis:
*   $H_0$: There is no difference in the mean AQI between Los Angeles County and the rest of California.
*   $H_A$: There is a difference in the mean AQI between Los Angeles County and the rest of California.


Set the significance level:


In [19]:
significant_level = 0.05
significant_level

0.05

Compute the p-value

In [21]:
stats.ttest_ind(a=ca_los['aqi'], b=ca_other['aqi'], equal_var=False)

TtestResult(statistic=2.1107010796372014, pvalue=0.049839056842410995, df=17.08246830361151)

With a p-value (0.049) being less than 0.05 (as your significance level is 5%), reject the null hypothesis in favor of the alternative hypothesis.

`With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?`

In [22]:
new = data[data['state_name']=='New York']
ohio = data[data['state_name']=='Ohio']

Formulate your null and alternative hypotheses:
*   $H_0$: The mean AQI of New York is greater than or equal to that of Ohio.
*   $H_A$: The mean AQI of New York is **below** that of Ohio.

Chose the previous significant level (5%)

Compute the p-value

In [23]:
stats.ttest_ind(a=new['aqi'], b=ohio['aqi'], alternative='less', equal_var=False)
# because the alternative find out below so the alternative is less

TtestResult(statistic=-2.025951038880333, pvalue=0.030446502691934683, df=15.036745051598716)

With a p-value (0.030) of less than 0.05 (as your significance level is 5%) and a t-statistic < 0 (-2.036), reject the null hypothesis in favor of the alternative hypothesis.

`A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?`

In [24]:
mic = data[data['state_name']=='Michigan']

Formulate your null and alternative hypotheses here:

*   $H_0$: The mean AQI of Michigan is less than or equal to 10.
*   $H_A$: The mean AQI of Michigan is greater than 10.

Chose the previous significant level (5%)

In [25]:
stats.ttest_1samp(mic['aqi'], 10, alternative='greater')
# Because the one sample so we just use ttest_1samp method

TtestResult(statistic=-1.7395913343286131, pvalue=0.9399405193140109, df=8)

With a p-value (0.940) being greater than 0.05 (as your significance level is 5%) and a t-statistic < 0 (-1.74), fail to reject the null hypothesis.